pkg/daemon/controller/node.go

```diff
@@ -78,63 +78,152 @@ func (r *nodeReconciler) Reconcile(ctx context.Context, request reconcile.Reques
 		return reconcile.Result{}, nil
 	}
 
+	vtepIP, vtepMac, err := r.selectVtepAddressFromLink()
+	if err != nil {
+		return reconcile.Result{Requeue: true}, fmt.Errorf("failed to select vtep address: %v", err)
+	}
+
+	vxlanLinkName, err := utils.GenerateVxlanNetIfName(r.ctrlHubRef.config.NodeVxlanIfName, overlayNetID)
+	if err != nil {
+		return reconcile.Result{Requeue: true}, fmt.Errorf("failed to generate vxlan interface name: %v", err)
+	}
+
+	thisNode := &corev1.Node{}
+	if err := r.Get(ctx, types.NamespacedName{Name: r.ctrlHubRef.config.NodeName}, thisNode); err != nil {
+		return reconcile.Result{Requeue: true}, fmt.Errorf("failed to get node %v object: %v",
+			r.ctrlHubRef.config.NodeName, err)
+	}
+
+	nodeLocalVxlanAddrs, err := r.selectNodeLocalVxlanAddrs(thisNode, vtepIP, vxlanLinkName)
+	if err != nil {
+		return reconcile.Result{Requeue: true}, fmt.Errorf("failed to select node local vxlan addresses: %v", err)
+	}
+
+	if err := r.updateNodeVxlanAnnotations(thisNode, vtepIP, vtepMac, nodeLocalVxlanAddrs); err != nil {
+		return reconcile.Result{Requeue: true}, fmt.Errorf("failed to update node vxlan annotations: %v", err)
+	}
+
+	nodeList := &corev1.NodeList{}
+	if err := r.List(ctx, nodeList); err != nil {
+		return reconcile.Result{Requeue: true}, fmt.Errorf("failed list node: %v", err)
+	}
+
+	vxlanDev, err := vxlan.NewVxlanDevice(vxlanLinkName, int(*overlayNetID),
+		r.ctrlHubRef.config.NodeVxlanIfName, vtepIP, r.ctrlHubRef.config.VxlanUDPPort,
+		r.ctrlHubRef.config.VxlanBaseReachableTime, true)
+	if err != nil {
+		return reconcile.Result{Requeue: true}, fmt.Errorf("failed to create vxlan device %v: %v", vxlanLinkName, err)
+	}
+
+	if err := ensureVxlanInterfaceAddresses(vxlanDev, nodeLocalVxlanAddrs); err != nil {
+		return reconcile.Result{Requeue: true}, fmt.Errorf("failed to ensure addresses for vxlan device %v: %v",
+			vxlanLinkName, err)
+	}
+
+	for _, node := range nodeList.Items {
+		if node.Annotations[constants.AnnotationNodeVtepMac] == "" ||
+			node.Annotations[constants.AnnotationNodeVtepIP] == "" ||
+			node.Annotations[constants.AnnotationNodeLocalVxlanIPList] == "" {
+			return reconcile.Result{Requeue: true}, fmt.Errorf("node %v's vtep information has not been updated", node.Name)
```

Comment:
is it better to skip this and aggregate all errors into one?

Reply:
If it's skipped, the logs might be flushed and make it harder to find the error. For the no-skipped version, there are two cases:
1. An error happens once start. The whole cluster will not be ready until error is fixed.
2. An error happens after something is running. New Pods will never get ready but exist Pods will not be effected.

I think that neither of the cases will bring accidents but only more efficiency of resolving problems.